runtime/connectors/s3/s3.go

```diff
@@ -192,3 +181,34 @@ func getAwsSessionConfig(ctx context.Context, conf *Config, bucket string) (*ses
 
 	return sess, nil
 }
+
+func resolvedCredentials(env *connectors.Env) (*credentials.Credentials, error) {
```

Comment:
Before, this logic was used only if no `S3Endpoint` was provided (but that may have been a mistake). Is it safe to use these credentials options for S3-compatible endpoints?

Reply:
Yes that was a mistake previously. Credential resolution is same for all